### PR TITLE
framework: Implement golang client/server

### PR DESCRIPTION
**Pull request by mestery**
This adds a refreshed golang-based client and server set of programs into the framework directory. We also build and push docker images for these client and server container images. Appropriate CI is added to GitHub Actions as well in the form of a new `framework-golang.yml` set of tests.

In the future, we can add the equivalent for the C++ based framework programs here.

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

**Comment by mestery** [comment]
The skips in the CI files are not right, correcting that now.

**Review by glimchb — APPROVED** [review]
looks reasonable, same as we did in OPI-POC repo for other impl

**Comment by mestery** [comment]
> looks reasonable, same as we did in OPI-POC repo for other impl

Exactly! The main difference is this API implements streaming, which was fun to implement here. We should look into that for our other APIs as well.

File: openoffload/go/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+version: '3.7'
```
Comment (seroyer) [inline openoffload/go/docker-compose.yml:1]: Probably want to add yml linter.  I think this will fail (missing ---).

**Pull request merged**